### PR TITLE
Port "root" level coffeescript to JS

### PR DESCRIPTION
**Pull request by naegelyd**
Progress towards #143.

### CODE REVIEW

File: varify/static/scripts/javascript/src/main.js [hunk 1]
```diff
@@ -1,135 +1,188 @@
...
-        });
-      }
-      if (c.isSupported('2.1.0')) {
-        data = {
-          queries: this.data.queries,
-          context: this.data.contexts.session,
-          view: this.data.views.session
+
+    var augmentFixedView = function() {
+        var new_view = {
```
Comment (bruth) [inline varify/static/scripts/javascript/src/main.js]: Not to nit.. but no camelCase?

File: varify/static/scripts/javascript/src/main.js [hunk 2]
```diff
@@ -1,135 +1,188 @@
...
+    // Force calls to the preview endpoint to use a fixed view composed solely
+    // of the sample concept. This will have the intended result of removing
+    // "duplicate" rows in the results table that sometimes occured due to
+    // the user's view.
+    c.config.set('session.defaults.data.preview', augmentFixedView);
+
+    // A simple handler for CONTEXT_REQUIRED and CONTEXT_INVALID events that
+    // tells the user which concept is required(when possible) or prints a
+    // generic message in the case the concept name could not be found.
+    var notify_required = (function(_this) {
```
Comment (bruth) [inline varify/static/scripts/javascript/src/main.js]: Camels..

File: varify/static/scripts/javascript/src/main.js [hunk 3]
```diff
@@ -1,135 +1,188 @@
...
-      } else {
-        message = 'There are 1 or more required concepts';
-      }
-      return c.notify({
-        level: 'error',
-        message: message
-      });
+}, [
+    'cilantro',
+    'project/ui',
```
Comment (bruth) [inline varify/static/scripts/javascript/src/main.js]: All the `project/` based paths could be relative paths. Using the `project/` usually only is needed on `require` calls or by external modules.

File: varify/static/scripts/javascript/src/templates.js [hunk 4]
```diff
@@ -1,283 +1,324 @@
...
+    };
+
+    var pathogenicity = function(assessment) {
+        if ((assessment != null) && (assessment.pathogenicity != null)) {
+            var html = [];
+            html.push('<br />Pathogenicity: ');
+            html.push(assessment.pathogenicity.name);
+            return '<span class=pathogenicity>' + (html.join('')) + '</span>';
+        }
+        else {
```
Comment (bruth) [inline varify/static/scripts/javascript/src/templates.js]: Somewhat of a preference thing, but `else` blocks after a return in the `if` block requires me to re-read the logic. I think this pattern is more readable.

``` javascript
if (...) {
    return '...';
}
return '';
```

File: varify/static/scripts/javascript/src/main.js (hunk 2 above)
Comment (naegelyd) [inline varify/static/scripts/javascript/src/main.js]: Ha, that's what I get for copying over translated coffee script...good
catch on both.

File: varify/static/scripts/javascript/src/templates.js [hunk 5]
```diff
@@ -1,283 +1,324 @@
...
+                    html.push('<tr class="hide no-border" id=assessment-row-' + assessment.id + '-details><td></td><td colspan=7><strong>Evidence Details: </strong>' + assessment.details + '</td></tr>');
+                }
+            }
+        });
+
+        return html.join('');
+    };
+
+    var hgmdLinks = function(phenotypes) {
+        var hgmd_phenotypes = [];
```
Comment (bruth) [inline varify/static/scripts/javascript/src/templates.js]: Camels.

File: varify/static/scripts/javascript/src/ui.js [hunk 6]
```diff
@@ -1,8 +1,15 @@
...
+    './ui/controls',
+    './ui/modals',
+    './ui/tables',
+    './ui/workflows'
+], function(_) {
+
+    // Modules to be mixed-in with exports
+    var mods = Array.prototype.slice.call(arguments, 1);
+
+    return _.extend.apply(null, mods);
```
Comment (bruth) [inline varify/static/scripts/javascript/src/ui.js]: Not that there are really any side effects, but this will extend the first module (controls) rather than a newly initialized object.

File: varify/static/scripts/javascript/src/templates.js (hunk 5 above)
Comment (penningtonj) [inline varify/static/scripts/javascript/src/templates.js]: What does this have to do with genome8ball.comhttp://genome8ball.com?

On Mar 20, 2014, at 7:48 PM, "Byron Ruth" <notifications@github.com<mailto:notifications@github.com>> wrote:

In varify/static/scripts/javascript/src/templates.js:

> -                html.push('<td>' + assessment.father_result + '</td>');
> -                html.push('<td>' + assessment.sanger + '</td>');
>   +
> -                html.push('</tr>');
>   +
> -                if (assessmentHasDetails) {
> -                    html.push('<tr class="hide no-border" id=assessment-row-' + assessment.id + '-details><td></td><td colspan=7><strong>Evidence Details: </strong>' + assessment.details + '</td></tr>');
> -                }
> -            }
> -        });
>   +
> -        return html.join('');
> -    };
>   +
> -    var hgmdLinks = function(phenotypes) {
> -        var hgmd_phenotypes = [];

Camels.

Comment (naegelyd) [inline varify/static/scripts/javascript/src/templates.js]: You forgot the "s"...
